test/unit/QueueConsumer-test.js

```diff
@@ -0,0 +1,314 @@
+/**
+ * QueueConsumer unit test
+ */
+require('require-self-ref')
+
+const test = require('ava')
+const sinon = require('sinon')
+
+const QueueConsumer = require('~/src/consumers/QueueConsumer')
+const { EventEmitter } = require('events')
+
+class MockChannel extends EventEmitter {
+  constructor () {
+    super()
+    this._closed = false
+  }
+
+  assertQueue (queueName, options) {
+    return Promise.resolve()
+  }
+
+  prefetch (prefetchValue) {
+    return Promise.resolve()
+  }
+
+  consume (queueName, onMessage) {
+    this.on('test-message', (testMessage) => {
+      onMessage(testMessage)
+    })
+  }
+
+  ack (message) {
+    return Promise.resolve()
+  }
+
+  nack (message) {
+    return Promise.resolve()
+  }
+
+  cancel (tag) {
+    return Promise.resolve()
+  }
+
+  close () {
+    if (this._closed) {
+      throw new Error('Already closed')
+    }
+    this._closed = true
+  }
+}
+
+class MockConnection extends EventEmitter {
+  constructor (channel) {
+    super()
+    this._mockChannel = channel
+  }
+
+  createChannel () {
+    return Promise.resolve(this._mockChannel)
+  }
+}
+
+const testQueueName = 'some-queue'
+const testMessage = { foo: 'bar' }
+
+test.beforeEach('setup mock channel and connections', (t) => {
+  const channel = new MockChannel(testQueueName)
+  const connection = new MockConnection(channel)
+  const consumer = new QueueConsumer({
+    queueName: testQueueName,
+    connection
+  })
+  t.context = { channel, connection, consumer }
+})
+
+test('should generate a _consumerTag based off of the queue name', (t) => {
+  const { consumer } = t.context
+  const consumerTag = consumer._consumerTag
+
+  t.true(consumerTag !== null)
+  t.true(consumerTag.startsWith(testQueueName))
+})
+
+/**
+ * Starting the queue consumer
+ */
+test('should should create a channel using the given connection', async (t) => {
+  const { channel, connection, consumer } = t.context
+  const mock = sinon.mock(connection)
```

Comment:
Thoughts on using http://sinonjs.org/releases/v2.3.5/sandbox/ in all tests? Helps avoid lingering stubs/spies/mocks with an afterEach restore.